test/DynamoCoreTests/CodeBlockNodeTests.cs

```diff
@@ -523,19 +523,78 @@ public void OutPort_WithCommentNonAssignment_Alignment()
 
             // The first output port should be at the first line
-            Assert.AreEqual(0 * Configurations.CodeBlockPortHeightInPixels, codeBlockNode.OutPortData[0].VerticalMargin, tolerance);
+            Assert.AreEqual(0, codeBlockNode.OutPortData[0].LineIndex);
 
             // The second output port should be at the 4th line, which is also 3 lines below the first
-            Assert.AreEqual(3 * Configurations.CodeBlockPortHeightInPixels, codeBlockNode.OutPortData[1].VerticalMargin, tolerance);
-            
+            Assert.AreEqual(4, codeBlockNode.OutPortData[1].LineIndex);
+
             code = "/*comment \n */ \n a[0]+b;";
             UpdateCodeBlockNodeContent(codeBlockNode, code);
 
             Assert.AreEqual(2, codeBlockNode.InPortData.Count);
             Assert.AreEqual(1, codeBlockNode.OutPortData.Count);
 
-            Assert.AreEqual(2 * Configurations.CodeBlockPortHeightInPixels, codeBlockNode.OutPortData[0].VerticalMargin, tolerance);
+            Assert.AreEqual(2, codeBlockNode.OutPortData[0].LineIndex);
+        }
+
+        [Test]
+        [Category("RegressionTests")]
+        public void PortIndicesShouldProduceCorrectConnectorOffsets()
+        {
+            var code =
+@"var00 = a;
+
+var01 = b;
+var02 = c;
+
+
+var03 = d;
+var04 = e;
+var05 = f;
+
+
+
+var06 = g;
+";
+
+            var codeBlockNode = CreateCodeBlockNode();
+            UpdateCodeBlockNodeContent(codeBlockNode, code);
 
+            Assert.AreEqual(7, codeBlockNode.InPortData.Count);
+            Assert.AreEqual(7, codeBlockNode.OutPortData.Count);
+
+            // Input ports are regular ports that do not depend on LineIndex.
+            Assert.AreEqual(-1, codeBlockNode.InPortData[0].LineIndex);
+            Assert.AreEqual(-1, codeBlockNode.InPortData[1].LineIndex);
+            Assert.AreEqual(-1, codeBlockNode.InPortData[2].LineIndex);
+            Assert.AreEqual(-1, codeBlockNode.InPortData[3].LineIndex);
+            Assert.AreEqual(-1, codeBlockNode.InPortData[4].LineIndex);
+            Assert.AreEqual(-1, codeBlockNode.InPortData[5].LineIndex);
+            Assert.AreEqual(-1, codeBlockNode.InPortData[6].LineIndex);
+
+            // Output ports are smaller ports that depend on LineIndex.
+            Assert.AreEqual(0, codeBlockNode.OutPortData[0].LineIndex);
+            Assert.AreEqual(2, codeBlockNode.OutPortData[1].LineIndex);
+            Assert.AreEqual(3, codeBlockNode.OutPortData[2].LineIndex);
+            Assert.AreEqual(6, codeBlockNode.OutPortData[3].LineIndex);
+            Assert.AreEqual(7, codeBlockNode.OutPortData[4].LineIndex);
+            Assert.AreEqual(8, codeBlockNode.OutPortData[5].LineIndex);
+            Assert.AreEqual(12, codeBlockNode.OutPortData[6].LineIndex);
+
+            // Ensure that "NodeModel.GetPortVerticalOffset" does not regress.
+            // This is the way connector position is calculated.
+            var verticalOffset = 2.9;
```

Comment:
`2.9` has been there in `GetPortVerticalOffset`, this is just to ensure the computation lined up nicely, otherwise either the test or `GetPortVerticalOffset` needs updating.